android-configure

```diff
@@ -8,72 +8,78 @@
 # modules with npm. Also, don't forget to set the arch in npm config using
 # 'npm config set arch=<arch>'
 
+if [ $# -ne 3 ]; then
+  echo "$0 should have 3 parameters: ndk_path, target_arch and sdk_version"
+  exit 1
+fi
+
+NDK_PATH=$1
+ARCH="$2"
+ANDROID_SDK_VERSION=$3
 
-if [ -z "$2" ]; then
-    ARCH=arm
-else
-    ARCH="$2"
+if [ $ANDROID_SDK_VERSION -lt 23 ]; then
+  echo "$ANDROID_SDK_VERSION should equal or later than 23(Android 6.0)"
 fi
 
 CC_VER="4.9"
+
 case $ARCH in
     arm)
-        DEST_CPU="$ARCH"
-        SUFFIX="$ARCH-linux-androideabi"
-        TOOLCHAIN_NAME="$SUFFIX"
+        DEST_CPU="arm"
+        TOOLCHAIN_NAME="armv7-linux-androideabi"
         ;;
     x86)
         DEST_CPU="ia32"
-        SUFFIX="i686-linux-android"
-        TOOLCHAIN_NAME="$ARCH"
+        TOOLCHAIN_NAME="i686-linux-android"
         ;;
     x86_64)
-        DEST_CPU="ia32"
-        SUFFIX="$ARCH-linux-android"
-        TOOLCHAIN_NAME="$ARCH"
+        DEST_CPU="x64"
+        TOOLCHAIN_NAME="x86_64-linux-android"
+        ARCH="x64"
+        ;;
+    arm64|aarch64)
+        DEST_CPU="arm64"
+        TOOLCHAIN_NAME="aarch64-linux-android"
+        ARCH="arm64"
         ;;
     *)
         echo "Unsupported architecture provided: $ARCH"
         exit 1
         ;;
 esac
 
-NDK_PATH=$1
-function make_toolchain {
-    $NDK_PATH/build/tools/make-standalone-toolchain.sh \
-         --toolchain=$TOOLCHAIN_NAME-$CC_VER \
-        --arch=$ARCH \
-        --install-dir=$TOOLCHAIN \
-        --platform=android-21
-}
+HOST_OS="linux"
+HOST_ARCH="x86_64"
+export CC_host=$(which gcc)
+export CXX_host=$(which g++)
 
-export TOOLCHAIN=$PWD/android-toolchain
-if [ -d "$TOOLCHAIN" ]; then
-    read -r -p "NDK toolchain already exists. Replace it?  [y/N]" response
-    case "$response" in
-        [Yy])
-            rm -rf "$TOOLCHAIN"
-            make_toolchain
-    esac
-else
-    make_toolchain
+host_gcc_version=$($CC_host --version | grep gcc | awk '{print $NF}')
+major=$(echo $host_gcc_version | awk -F . '{print $1}')
+minor=$(echo $host_gcc_version | awk -F . '{print $2}')
+if [ -z $major ] || [ -z $minor ] || [ $major -lt 6 ] || [ $minor -lt 3 ]; then
```

Comment:
Won't this fail on something like gcc 8.1 (major > 6 but minor < 3)?

Reply:
thanks, it will fail on this case. uploaded